contrib/peer-logging/peer-logging-parser.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Parse message log binary files.  To be used in conjunction with -logmessages."""
+
+import argparse
+import os
+import sys
+from io import BytesIO
+import json
+from pathlib import Path
+from typing import Any, List
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../test/functional'))
+
+from test_framework.mininode import MESSAGEMAP  # noqa: E402
+
+TIME_SIZE = 8
+LENGTH_SIZE = 4
+COMMAND_SIZE = 12
+
+def to_jsonable(obj: Any) -> Any:
+    ret = {}    # type: Any
+    # The msg objects don't generally include "msgtype" in their __slots__
+    if hasattr(obj, "msgtype"):
+        ret["msgtype"] = getattr(obj, "msgtype", None).decode()
+    if hasattr(obj, "__dict__"):
+        ret = obj.__dict__
+    elif hasattr(obj, "__slots__"):
+        for slot in obj.__slots__:
+            val = getattr(obj, slot, None)
+            ret[slot] = to_jsonable(val)
+    elif isinstance(obj, list):
+        ret = [to_jsonable(a) for a in obj]
+    elif isinstance(obj, bytes):
+        ret = obj.hex()
+    else:
+        ret = obj
+    return ret
+
+
+def process_file(path: Path, messages: List[Any], recv: bool) -> None:
+    with open(path, 'rb') as f_in:
+        while True:
+            tmp_header_raw = f_in.read(TIME_SIZE + LENGTH_SIZE + COMMAND_SIZE)
+            if not tmp_header_raw:
+                break
+            tmp_header = BytesIO(tmp_header_raw)
+            time = int.from_bytes(tmp_header.read(TIME_SIZE), "little") # type: int
+            command = tmp_header.read(COMMAND_SIZE).split(b'\x00', 1)[0]  # type: bytes
+            length = int.from_bytes(tmp_header.read(LENGTH_SIZE), "little") # type: int
+            if command not in MESSAGEMAP:
+                continue    # For now just skip unrecognized messages
+            msg = MESSAGEMAP[command]()
+            msg.deserialize(f_in)
+            msg_dict = to_jsonable(msg)
+            msg_dict["time"] = time
+            msg_dict["length"] = length
+            msg_dict["recv"] = recv
```

Comment:
`"recv": true` and `"recv": false` don't seem very friendly here. Perhaps `"direction": "received"` and `"direction": "sent"` is better?

Reply:
I'll meet you in the middle and do `"direction": "recv"` and `"direction": "sent"` :)